app/src/test/java/com/jalasoft/routesapp/util/algorithm/RouteCalculatorTest.kt

```diff
@@ -0,0 +1,75 @@
+package com.jalasoft.routesapp.util.algorithm
+
+import com.jalasoft.routesapp.data.model.remote.CityRouteForAlgorithm
+import com.jalasoft.routesapp.data.model.remote.Route
+import com.jalasoft.routesapp.data.model.remote.TransportMethod
+import com.jalasoft.routesapp.data.source.CountriesFakedata
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.HiltTestApplication
+import junit.framework.TestCase
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+@Config(application = HiltTestApplication::class, manifest = Config.NONE)
+@HiltAndroidTest
+class RouteCalculatorTest : TestCase() {
+    @get:Rule
+    val hiltRule = HiltAndroidRule(this)
+
+    lateinit var cityRouteForAlgorithm: CityRouteForAlgorithm
+
+    @Before
+    public override fun setUp() {
+        super.setUp()
+        val start1 = RouteCalculator.coordinatesToLocation(-16.52035351419114, -68.12580890707301)
```

Comment:
tests are good but as you will need some refactor on the method, the tests will be affected too.

Reply:
Changed